DOCS-7334 API Groups (#106)
mulesoft/docs-exchange: modules/ROOT/pages/to-create-an-asset.adoc

:imagesdir: ../assets/images/
:page-aliases: to-publish-an-asset.adoc

> In Anypoint Exchange private and Exchange portal, to share an API, API Group, policy, example, template, or connector, create an asset of that asset type.

Reviewer: Ditto re. these names....
Given context, could this be...:
```suggestion
To share an API, API Group, policy, example, template, or connector in either local or public Exchange, create an asset of that asset type.
```

Author: Fixed.

== Create an API Group Asset

> An API Group is an asset that enables organizations to publish a group of API instances as a single unit, so developers' client applications can access them as a group, using one client ID and client secret.

Reviewer: ```suggestion
An API Group is an asset that enables organizations to publish a group of API instances as a single unit, so that  client applications can access the APIs as a group, using one client ID and, optionally, client secret.
```

Author: Fixed.

> An API Group is an asset that enables organizations to publish a group of related APIs as a single unit, so developers' client applications can access that group as a single unit, with a single client ID and client secret.

Reviewer 2: Delete the word "related" and include the word instances, because that is what is really being grouped.

Author: Fixed.

> An API Group is an asset that enables organizations to publish a group of related APIs as a single unit, so developers' client applications can access that group as a single unit, with a single client ID and client secret.

Reviewer 3: ```diff
- with a single client ID and client secret
+ with the the same client ID and client secret
```
(Just because there was too many uses of word `single`)

Author: Good point! Fixed.

API Groups are created in Anypoint API Manager and published to Exchange.

> API Groups have major versions such as 1.0.0, 2.0.0, or 3.0.0, but no minor or patch versions. Every version of an API Group has one or more API Group instances. Each API Group instance is a group of API instances that all have the same identity provider and the same environment type, either production or sandbox. An API Group instance can contain API instances from multiple business groups.

Reviewer: ```suggestion
API Groups have major versions, such as 1.0.0, 2.0.0, or 3.0.0, but no minor or patch versions. Every version of an API Group has one or more API Group instances. Each API Group instance is a group of API instances that all have the same identity provider and the same environment type, either production or sandbox. An API Group instance can contain API instances from multiple business groups.
```

> Each API Group can have multiple versions. Every major version of an API Group has one or more API Group instances. Each API Group instance is a group of API instances that are all in the same environment, such as a development, testing, or production environment, and that all have the same identity provider. An API Group instance can contain API instances from multiple business groups.

Reviewer 2: Suggest removing the word "Major". >> ... Every version of an API Group has one or more API Group instances.

Include Type >> Each API Group instance is a group of API instances that are all in the same environment type, such as ...

Author: Much better! Fixed.

> Each API Group can have multiple versions. Every major version of an API Group has one or more API Group instances. Each API Group instance is a group of API instances that are all in the same environment, such as a development, testing, or production environment, and that all have the same identity provider. An API Group instance can contain API instances from multiple business groups.

Reviewer 3: "Each API Group can have multiple versions"

I think we can be more precise here with:

> An API Group can have different versions as any other asset type, but it only has major versions (1.0.0, 2.0.0, and so on)

Reviewer 3: > Each API Group instance is a group of API instances that are all in the same environment

They all share the same `environment type`, but they can come from different environments.
There are two types of environments: Production and Sandbox

![image](https://user-images.githubusercontent.com/23746912/83299671-00158780-a1cd-11ea-9bd4-8b487b5e79b1.png)
(You also have `Design`, but you can't create instances in it)

Author: Much better, and thank you for the correction about environment types! Fixed.

Creating an API Group requires having the group administrator role, and the asset administrator role for each of the APIs in the group. This ensures that a group creator can change the underlying APIs between private and public visibility levels.

> Every API instance in an API Group instance can be accessed through the API Group instance, and this access can be granted on the API Group instance contracts page, without approval on the API instance contracts page. You can view the group contracts for an API on the API's contracts page.

Reviewer 3: I don' understand this. I think we can better leave this part to API Manager documentation, they should explain how the API Group instance owner can grant acces and see contracts.

Author: Agreed. I'll delete this and let the API Manager docs explain it.

> If an API Group has any private APIs, you see a warning when publishing the API Group to your public portal. The public portal never shows private content. To ensure that all content in the API Group is published to the public portal, go to each API's page in Exchange and make the API public, and then publish the API Group.

Reviewer: GLOBALLY standardize: private or local?

Reviewer: Wait, have we clarified under what circumstances you can make local/private assets public? Aren't they private for a reason?

Author: The global standard is private and not local. What sparked the confusion? Was "local" in the docs somewhere?

Typically an org will have private assets in development and then make them public to release them to external devs, which is covered in the "Share Assets" page.

> If the API Group has any API content that is private, such as instances, pages, or terms and conditions, you see a warning when publishing the API Group to your public portal. Private content is not published to the public portal. To ensure that all content in the API Group is published to the public portal, go to each API's page in Exchange and make the API public, and then publish the API Group.

Reviewer 3: ```diff
- Private content is not published to the public portal
+ Private content is never shown in the public portal
```

Author: Even better. Fixed.

> If the API Group has any API content that is private, such as instances, pages, or terms and conditions, you see a warning when publishing the API Group to your public portal. Private content is not published to the public portal. To ensure that all content in the API Group is published to the public portal, go to each API's page in Exchange and make the API public, and then publish the API Group.

Reviewer 2: Warnings only appear if:
- API Group is published, and at least one underlying APIs is private. (only visible on share modal)
- API Group is published, and none instance group is public.
- If an instance group is made "Public" and all its instances are private.

Author: Very useful to learn! Fixed.
